### PR TITLE
Import repair

### LINKED ISSUE #1064: Versioning Issues
https://github.com/cosmos/gaia/blob/ad20b3f37db8de87dc27f64310a01ba6ddede35d/go.sum#L1002-L1005

### PR DESCRIPTION
**Pull request by faddat**
Closes: #1064 

If we ship current v6.0.0 we'll ship with an iavl that just had race conditions identified on it.

Furthermore, we'd be shipping with a go.sum that imports like 4 versions of iavl.  

And we'd be shipping it with Tendermint as far back as v0.34.0-rc0

Personally, I just prefer that we don't do that. 

This PR also makes CI use the latest golangci-lint by default, and changes some settings that were making ci fail due to the deprecation of various linters. 

You can think of this PR as mainly problem-identification and defense.  The real solution here is better coordination throughout the cosmos ecosystem.

Here's a tweet where I summarized findings, and also linked to pull requests that fully resolve the issue and would make the replace statements at the bottom of go.mod unnecessary. 

twitter.com/gadikian/status/1469512833809678342

If we wanted to ship as-is everything would most likely be fine.  Also, there's a slim chance that my changes actually introduce issues by causing imported modules to use incompatible versions of software, but so far, so good. 

### Summary of findings:

* Cosmos-SDK v0.44.3 pulls in tendermint as old as v0.34.0-rc4, as well as a tm-db v0.5.* or two and probably shouldn't be used because of this. 

* Cosmos-SDK v0.44.5 does not pull in old-tendermint and also does not pull in old tm-db, but should make a release that uses tm-db v0.6.6 in the v0.44.* line. 

* IBC-go v2.0.0 pulls in cosmos-sdk v0.44.3 and is fixed here:
https://github.com/cosmos/ibc-go/pull/611

* the liquidity module pulls in cosmos-sdk v0.44.3 and is fixed here:
https://github.com/Gravity-Devs/liquidity/pull/7

* the ibc router pulls in cosmos-sdk v0.44.3 and is fixed here:
https://github.com/strangelove-ventures/packet-forward-middleware/pull/5

So, if those three PRs were merged pre-launch, I think that'd be a step towards an easier to maintain, possibly more performant, possibly more secure gaia.

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -189,14 +159,11 @@ github.com/cockroachdb/datadriven v0.0.0-20190809214429-80d97fb3cbaa/go.mod h1:z
 github.com/codahale/hdrhistogram v0.0.0-20161010025455-3a0bb77429bd/go.mod h1:sE/e/2PUdi/liOCUjSTXgM1o87ZssimdTWN964YiIeI=
 github.com/coinbase/rosetta-sdk-go v0.6.10 h1:rgHD/nHjxLh0lMEdfGDqpTtlvtSBwULqrrZ2qPdNaCM=
 github.com/coinbase/rosetta-sdk-go v0.6.10/go.mod h1:J/JFMsfcePrjJZkwQFLh+hJErkAmdm9Iyy3D5Y0LfXo=
-github.com/confio/ics23/go v0.0.0-20200817220745-f173e6211efb/go.mod h1:E45NqnlpxGnpfTWL/xauN7MRwEE28T4Dd4uraToOaKg=
-github.com/confio/ics23/go v0.6.3/go.mod h1:E45NqnlpxGnpfTWL/xauN7MRwEE28T4Dd4uraToOaKg=
 github.com/confio/ics23/go v0.6.6 h1:pkOy18YxxJ/r0XFDCnrl4Bjv6h4LkBSpLS6F38mrKL8=
 github.com/confio/ics23/go v0.6.6/go.mod h1:E45NqnlpxGnpfTWL/xauN7MRwEE28T4Dd4uraToOaKg=
```
Comment (faddat) [inline go.sum:181]: this

File: go.sum [hunk 2]
```diff
@@ -205,22 +172,23 @@ github.com/coreos/go-semver v0.2.0/go.mod h1:nnelYz7RCh+5ahJtPPxZlU+153eP4D4r3Ee
...
-github.com/cosmos/iavl v0.15.0-rc3.0.20201009144442-230e9bdf52cd/go.mod h1:3xOIaNNX19p0QrX0VqWa6voPRoJRGGYtny+DH8NEPvE=
-github.com/cosmos/iavl v0.15.0-rc5/go.mod h1:WqoPL9yPTQ85QBMT45OOUzPxG/U/JcJoN7uMjgxke/I=
-github.com/cosmos/iavl v0.15.3/go.mod h1:OLjQiAQ4fGD2KDZooyJG9yz+p2ao2IAYSbke8mVvSA4=
-github.com/cosmos/iavl v0.17.1 h1:b/Cl8h1PRMvsu24+TYNlKchIu7W6tmxIBGe6E9u2Ybw=
-github.com/cosmos/iavl v0.17.1/go.mod h1:7aisPZK8yCpQdy3PMvKeO+bhq1NwDjUwjzxwwROUxFk=
-github.com/cosmos/ibc-go/v2 v2.0.0 h1:BMRg73JcdV9wGPI51j89ihm7VBZQsDLkqQ+tmzdeA9Y=
+github.com/cosmos/gorocksdb v0.0.0-20211202124722-2c356d6d98e4 h1:WxvYoiQOATTS4dNsqEKiRWGNPJHs5BM2cbO74NlhkUo=
+github.com/cosmos/gorocksdb v0.0.0-20211202124722-2c356d6d98e4/go.mod h1:b/U29r/CtguX3TF7mKG1Jjn4APDqh4wECshxXdiWHpA=
+github.com/cosmos/iavl v0.17.3 h1:s2N819a2olOmiauVa0WAhoIJq9EhSXE9HDBAoR9k+8Y=
+github.com/cosmos/iavl v0.17.3/go.mod h1:prJoErZFABYZGDHka1R6Oay4z9PrNeFFiMKHDAMOi4w=
```
Comment (faddat) [inline go.sum:203]: this

File: go.sum [hunk 3]
```diff
@@ -205,22 +172,23 @@ github.com/coreos/go-semver v0.2.0/go.mod h1:nnelYz7RCh+5ahJtPPxZlU+153eP4D4r3Ee
...
 github.com/cosmos/go-bip39 v1.0.0 h1:pcomnQdrdH22njcAatO0yWojsUnCO3y2tNoV1cb6hHY=
 github.com/cosmos/go-bip39 v1.0.0/go.mod h1:RNJv0H/pOIVgxw6KS7QeX2a0Uo0aKUlfhZ4xuwvCdJw=
-github.com/cosmos/iavl v0.15.0-rc3.0.20201009144442-230e9bdf52cd/go.mod h1:3xOIaNNX19p0QrX0VqWa6voPRoJRGGYtny+DH8NEPvE=
-github.com/cosmos/iavl v0.15.0-rc5/go.mod h1:WqoPL9yPTQ85QBMT45OOUzPxG/U/JcJoN7uMjgxke/I=
-github.com/cosmos/iavl v0.15.3/go.mod h1:OLjQiAQ4fGD2KDZooyJG9yz+p2ao2IAYSbke8mVvSA4=
-github.com/cosmos/iavl v0.17.1 h1:b/Cl8h1PRMvsu24+TYNlKchIu7W6tmxIBGe6E9u2Ybw=
-github.com/cosmos/iavl v0.17.1/go.mod h1:7aisPZK8yCpQdy3PMvKeO+bhq1NwDjUwjzxwwROUxFk=
-github.com/cosmos/ibc-go/v2 v2.0.0 h1:BMRg73JcdV9wGPI51j89ihm7VBZQsDLkqQ+tmzdeA9Y=
+github.com/cosmos/gorocksdb v0.0.0-20211202124722-2c356d6d98e4 h1:WxvYoiQOATTS4dNsqEKiRWGNPJHs5BM2cbO74NlhkUo=
+github.com/cosmos/gorocksdb v0.0.0-20211202124722-2c356d6d98e4/go.mod h1:b/U29r/CtguX3TF7mKG1Jjn4APDqh4wECshxXdiWHpA=
```
Comment (faddat) [inline go.sum]: this

File: go.sum [hunk 4]
```diff
@@ -205,22 +172,23 @@ github.com/coreos/go-semver v0.2.0/go.mod h1:nnelYz7RCh+5ahJtPPxZlU+153eP4D4r3Ee
...
-github.com/cosmos/cosmos-sdk v0.44.2 h1:EWoj9h9Q9t7uqS3LyqzZWWwnSEodUJlYDMloDoPBD3Y=
-github.com/cosmos/cosmos-sdk v0.44.2/go.mod h1:fwQJdw+aECatpTvQTo1tSfHEsxACdZYU80QCZUPnHr4=
+github.com/cosmos/btcutil v1.0.4 h1:n7C2ngKXo7UC9gNyMNLbzqz7Asuf+7Qv4gnX/rOdQ44=
+github.com/cosmos/btcutil v1.0.4/go.mod h1:Ffqc8Hn6TJUdDgHBwIZLtrLQC1KdJ9jGJl/TvgUaxbU=
+github.com/cosmos/cosmos-sdk v0.44.5 h1:t5h+KPzZb0Zsag1RP1DCMQlyJyIQqJcqSPJrbUCDGHY=
+github.com/cosmos/cosmos-sdk v0.44.5/go.mod h1:maUA6m2TBxOJZkbwl0eRtEBgTX37kcaiOWU5t1HEGaY=
 github.com/cosmos/go-bip39 v0.0.0-20180819234021-555e2067c45d/go.mod h1:tSxLoYXyBmiFeKpvmq4dzayMdCjCnu8uqmCysIGBT2Y=
 github.com/cosmos/go-bip39 v1.0.0 h1:pcomnQdrdH22njcAatO0yWojsUnCO3y2tNoV1cb6hHY=
 github.com/cosmos/go-bip39 v1.0.0/go.mod h1:RNJv0H/pOIVgxw6KS7QeX2a0Uo0aKUlfhZ4xuwvCdJw=
-github.com/cosmos/iavl v0.15.0-rc3.0.20201009144442-230e9bdf52cd/go.mod h1:3xOIaNNX19p0QrX0VqWa6voPRoJRGGYtny+DH8NEPvE=
```
Comment (faddat) [inline go.sum:217]: ?

File: go.sum [hunk 5]
```diff
@@ -205,22 +172,23 @@ github.com/coreos/go-semver v0.2.0/go.mod h1:nnelYz7RCh+5ahJtPPxZlU+153eP4D4r3Ee
...
-github.com/cosmos/iavl v0.15.0-rc5/go.mod h1:WqoPL9yPTQ85QBMT45OOUzPxG/U/JcJoN7uMjgxke/I=
-github.com/cosmos/iavl v0.15.3/go.mod h1:OLjQiAQ4fGD2KDZooyJG9yz+p2ao2IAYSbke8mVvSA4=
```
Comment (faddat) [inline go.sum:219]: this

File: go.sum [hunk 6]
```diff
@@ -205,22 +172,23 @@ github.com/coreos/go-semver v0.2.0/go.mod h1:nnelYz7RCh+5ahJtPPxZlU+153eP4D4r3Ee
...
-github.com/cosmos/iavl v0.17.1 h1:b/Cl8h1PRMvsu24+TYNlKchIu7W6tmxIBGe6E9u2Ybw=
-github.com/cosmos/iavl v0.17.1/go.mod h1:7aisPZK8yCpQdy3PMvKeO+bhq1NwDjUwjzxwwROUxFk=
```
Comment (faddat) [inline go.sum:221]: this

File: go.sum [hunk 7]
```diff
@@ -990,89 +800,53 @@ github.com/subosito/gotenv v1.2.0/go.mod h1:N0PQaV/YGNqwC0u51sEeR/aUtSLEXKX9iv69
...
-github.com/tdakkota/asciicheck v0.0.0-20200416200610-e657995f937b/go.mod h1:yHp0ai0Z9gUljN3o0xMhYJnH/IcvkdTBOX2fmJ93JEM=
-github.com/tecbot/gorocksdb v0.0.0-20191217155057-f0fad39f321c h1:g+WoO5jjkqGAzHWCjJB1zZfXPIAaDpzXIEJ0eS6B5Ok=
-github.com/tecbot/gorocksdb v0.0.0-20191217155057-f0fad39f321c/go.mod h1:ahpPrc7HpcfEWDQRZEmnXMzHY03mLDYMCxeDzy46i+8=
 github.com/tendermint/btcd v0.1.1 h1:0VcxPfflS2zZ3RiOAHkBiFUcPvbtRj5O7zHmcJWHV7s=
 github.com/tendermint/btcd v0.1.1/go.mod h1:DC6/m53jtQzr/NFmMNEu0rxf18/ktVoVtMrnDD5pN+U=
 github.com/tendermint/crypto v0.0.0-20191022145703-50d29ede1e15 h1:hqAk8riJvK4RMWx1aInLzndwxKalgi5rTqgfXxOxbEI=
 github.com/tendermint/crypto v0.0.0-20191022145703-50d29ede1e15/go.mod h1:z4YtwM70uOnk8h0pjJYlj3zdYwi9l03By6iAIF5j/Pk=
 github.com/tendermint/go-amino v0.16.0 h1:GyhmgQKvqF82e2oZeuMSp9JTN0N09emoSZlb2lyGa2E=
 github.com/tendermint/go-amino v0.16.0/go.mod h1:TQU0M1i/ImAo+tYpZi73AU3V/dKeCoMC9Sphe2ZwGME=
-github.com/tendermint/tendermint v0.34.0-rc4/go.mod h1:yotsojf2C1QBOw4dZrTcxbyxmPUrT4hNuOQWX9XUwB4=
```
Comment (faddat) [inline go.sum:1002]: ?

File: go.sum [hunk 8]
```diff
@@ -990,89 +800,53 @@ github.com/subosito/gotenv v1.2.0/go.mod h1:N0PQaV/YGNqwC0u51sEeR/aUtSLEXKX9iv69
...
-github.com/tendermint/tendermint v0.34.0-rc6/go.mod h1:ugzyZO5foutZImv0Iyx/gOFCX6mjJTgbLHTwi17VDVg=
-github.com/tendermint/tendermint v0.34.0/go.mod h1:Aj3PIipBFSNO21r+Lq3TtzQ+uKESxkbA3yo/INM4QwQ=
```
Comment (faddat) [inline go.sum:1004]: this

File: go.sum [hunk 9]
```diff
@@ -990,89 +800,53 @@ github.com/subosito/gotenv v1.2.0/go.mod h1:N0PQaV/YGNqwC0u51sEeR/aUtSLEXKX9iv69
...
-github.com/tendermint/tendermint v0.34.13/go.mod h1:6RVVRBqwtKhA+H59APKumO+B7Nye4QXSFc6+TYxAxCI=
 github.com/tendermint/tendermint v0.34.14 h1:GCXmlS8Bqd2Ix3TQCpwYLUNHe+Y+QyJsm5YE+S/FkPo=
 github.com/tendermint/tendermint v0.34.14/go.mod h1:FrwVm3TvsVicI9Z7FlucHV6Znfd5KBc/Lpp69cCwtk0=
-github.com/tendermint/tm-db v0.6.2/go.mod h1:GYtQ67SUvATOcoY8/+x6ylk8Qo02BQyLrAs+yAcLvGI=
-github.com/tendermint/tm-db v0.6.3/go.mod h1:lfA1dL9/Y/Y8wwyPp2NMLyn5P5Ptr/gvDFNWtrCWSf8=
 github.com/tendermint/tm-db v0.6.4 h1:3N2jlnYQkXNQclQwd/eKV/NzlqPlfK21cpRRIx80XXQ=
 github.com/tendermint/tm-db v0.6.4/go.mod h1:dptYhIpJ2M5kUuenLr+Yyf3zQOv1SgBZcl8/BmWlMBw=
```
Comment (faddat) [inline go.sum:890]: this

File: go.sum [hunk 10]
```diff
@@ -990,89 +800,53 @@ github.com/subosito/gotenv v1.2.0/go.mod h1:N0PQaV/YGNqwC0u51sEeR/aUtSLEXKX9iv69
...
 github.com/yuin/goldmark v1.1.32/go.mod h1:3hX8gzYuyVAZsxl0MRgGTJEmQBFcNTphYh9decYSb74=
 github.com/yuin/goldmark v1.2.1/go.mod h1:3hX8gzYuyVAZsxl0MRgGTJEmQBFcNTphYh9decYSb74=
 github.com/yuin/goldmark v1.3.5/go.mod h1:mwnBkeHKe2W/ZEtQ+71ViKU8L12m81fl3OWwC1Zlc8k=
 github.com/zondax/hid v0.9.0 h1:eiT3P6vNxAEVxXMw66eZUAAnU2zD33JBkfG/EnfAKl8=
 github.com/zondax/hid v0.9.0/go.mod h1:l5wttcP0jwtdLjqjMMWFVEE7d1zO0jvSPA9OPZxWpEM=
-go.etcd.io/bbolt v1.3.2/go.mod h1:IbVyRI1SCnLcuJnV2u8VeU0CEYM7e686BmAb1XKL+uU=
-go.etcd.io/bbolt v1.3.3/go.mod h1:IbVyRI1SCnLcuJnV2u8VeU0CEYM7e686BmAb1XKL+uU=
-go.etcd.io/bbolt v1.3.4/go.mod h1:G5EMThwa9y8QZGBClrRx5EY+Yw9kAhnjy3bSjsnlVTQ=
 go.etcd.io/bbolt v1.3.5 h1:XAzx9gjCb0Rxj7EoqcClPD1d5ZBxZJk0jbuoPHenBt0=
 go.etcd.io/bbolt v1.3.5/go.mod h1:G5EMThwa9y8QZGBClrRx5EY+Yw9kAhnjy3bSjsnlVTQ=
```
Comment (faddat) [inline go.sum:930]: this

**Comment by faddat** [comment]
There are two solutions to this, and really I figure that it's best that for now we pursue both:

* replace statements
* collaboration in the ecosystem to ensure that we do not import ancient versions of our own software

It's time for sleep, but I can fix the stuff the linter found tomorrow.

**Comment by faddat** [comment]
https://github.com/CosmWasm/wasmd/pull/692

**Comment by faddat** [comment]
I checked through cosmos-sdk 0.44.3 and 0.44.5.  While 0.44.3 exhibits this issue, 0.44.5 does not.

**Comment by faddat** [comment]
Here's a full summary-- including upstream PR's for any and all repositories that cause the old iavl imports. 

https://twitter.com/gadikian/status/1469512833809678342

**Comment by faddat** [comment]
This patch now satisfies the linter, too :)

**Comment by tac0turtle** [comment]
Hey could you run `go mod tidy` . Having multiple checksums does not mean we are importing multiple versions -> https://go.dev/ref/mod#minimal-version-selection

**Comment by faddat** [comment]
on this, or on gaia's main branch?

**Comment by tac0turtle** [comment]
this

**Comment by faddat** [comment]
sure :)

**Comment by faddat** [comment]
Screen Shot 2021-12-12 at 7 44 20 PM

exactly like that?

File: go.sum (hunk 6 above)
Comment (faddat) [reply go.sum:221]: @marbar3778 this is of specific concern

File: go.sum (hunk 8 above)
Comment (faddat) [reply go.sum:1004]: @marbar3778 this too

**Comment by faddat** [comment]
@marbar3778 what I think is happening is that various things that gaia imports, pull in really old software, and build against them, and that's my concern here.

**Comment by tac0turtle** [comment]
In gaia the bot that we had fixing this was removed, still unclear as to why? I think if we add depndabot back for go modules it should keep modules and versions up to date.

**Comment by faddat** [comment]
I should mention that this problem is **very much not limited to gaia**.  

I think I am up to six chains, and I think I could easily find it to be near-universal. 

See, thing is, it isn't enough to just specify stuff in go.mod because a lot of this comes from the cosmos-sdk.  v0.44.3 shows this problem, while 0.44.5 doesn't.  

Then, if you import anything that depends on cosmos-sdk v0.44.3 or lower, you end up getting tendermint 0.34.0-rc4 in go.sum and -- please correct me if I'm wrong-- that's no bueno.  I am going to try this: `go list-m all`

```
MVS starts at the main module (a special vertex in the graph that has no version) and traverses the graph, tracking the highest required version of each module. At the end of the traversal, the highest required versions comprise the build list: they are the minimum versions that satisfy all requirements.

The build list may be inspected with the command go list -m all. Unlike other dependency management systems, the build list is not saved in a “lock” file. MVS is deterministic, and the build list doesn’t change when new versions of dependencies are released, so MVS is used to compute it at the beginning of every module-aware command.

Consider the example in the diagram below. The main module requires module A at version 1.2 or higher and module B at version 1.2 or higher. A 1.2 and B 1.2 require C 1.3 and C 1.4, respectively. C 1.3 and C 1.4 both require D 1.2.
```



Then, the problem comes back into the main branch of cosmos-sdk here:

https://github.com/cosmos/cosmos-sdk/pull/10570

**Comment by faddat** [comment]
Well, well.....

@marbar3778 thanks for the teachin.  Can you let me know if -- 

1) this is actually an issue, despite what this command says below

or 

2) I was chasing a nothingburger because Go was showing the lowest version that it could build against, then building against a required version?

or

3) something else, which I still don't understand?


main:



```
 ~/testiepoo (main)> go list -m all | grep tendermint
github.com/tendermint/btcd v0.1.1
github.com/tendermint/crypto v0.0.0-20191022145703-50d29ede1e15
github.com/tendermint/go-amino v0.16.0
github.com/tendermint/tendermint v0.34.14
github.com/tendermint/tm-db v0.6.4
```

```
~/testiepoo (main)> go list -m all | grep cosmos
github.com/cosmos/gaia/v6
github.com/cosmos/cosmos-sdk v0.44.3 => github.com/cosmos/cosmos-sdk v0.44.2
github.com/cosmos/go-bip39 v1.0.0
github.com/cosmos/iavl v0.17.1
github.com/cosmos/ibc-go/v2 v2.0.0
github.com/cosmos/ledger-cosmos-go v0.11.1
github.com/cosmos/ledger-go v0.9.2
github.com/regen-network/cosmos-proto v0.3.1
```

**Comment by williambanfield** [comment]
Hey @faddat, Thanks for taking a look into the dependencies of Gaia, it's really helpful to have many eyes on this project. 

The behavior we appear to be seeing here is summarized in the `go.sum` [reference](https://go.dev/ref/mod#go-sum-files). Namely: 
```
The go.sum file may contain hashes for multiple versions of a module. The go command may need to load go.mod files from multiple versions of a dependency in order to perform minimal version selection. 
```
This would make sense, since the hashes in the `go.sum` that correspond to old versions correspond to the hashes of `go.mod` files, not that hash of the library themselves. This is because `go mod` needed to download the mod file while computing the MVS. You can similarly see that it has done that for `cloud.google.com/go` at the top of the `go.sum` here (note the `/go.mod` at the end of the version). The list of versions that will be included in any build can be found by running `go list -m all`, and in the case of gaia, reports `github.com/cosmos/iavl v0.17.1` and`github.com/tendermint/tendermint v0.34.14`.

**Comment by williambanfield** [comment]
Hey @faddat, it looks like what we're seeing here is this [behavior](https://go.dev/ref/mod#go-sum-files) of `go.sum`:

```
The go.sum file may contain hashes for multiple versions of a module. The go command may need to load go.mod files from multiple versions of a dependency in order to perform minimal version selection.
```

You can see that a similar stanza exists at the top of the `go.sum` file containing many different `go.mod` files correspond to `cloud.google.com/go` versions. These are stored because the tooling fetched them and used them as part of its calculation for MVS. 

The version that the binary will be built with is the version output by `go list -m all`

**Comment by faddat** [comment]
Thanks @marbar3778, @williambanfield -- final question-- the difference in go.sum between cosmos-sdk 0.44.3 and 0.44.5 -- also nothing?  (that difference, is really quite vast, but it seems I've learned a great deal about Go today)

```
cloud.google.com/go
```
refers to `go` itself?

Thanks again!

**Comment by faddat** [comment]
Hey, please let me know if this should be closed, left open, or what-- pretty clear that I was at least to some degree chasing a nothingburger based on a misnterpertation of go.sum

**Comment by tac0turtle** [comment]
The dependency update should be merged imo.

**Comment by williambanfield** [comment]
> Thanks @marbar3778, @williambanfield -- final question-- the difference in go.sum between cosmos-sdk 0.44.3 and 0.44.5 -- also nothing? (that difference, is really quite vast, but it seems I've learned a great deal about Go today)

Correct, the `go.sum` file does not dictate what's going to be included in a build. It's just some record keeping the tooling does to ensure that it can fetch a consistent version of a dependency. It's effectively a checksum, not a lock file. It has no effect on what is in the dependency graph. It should generally be ignored. 

> 
> ```
> cloud.google.com/go
> ```
> 
> refers to `go` itself?
> 
> Thanks again!

From `go mod graph`, that shows as a dependency of a grpc library:

```
...
google.golang.org/grpc@v1.19.1 cloud.google.com/go@v0.26.0
```

[Repo here](https://github.com/googleapis/google-cloud-go). 

Thanks for keeping an eye on our imports. If you do spot issues where our versions look out of date, please feel free to open an issue or reach out.

**Comment by williambanfield** [comment]
I think we can close this for now. The `replace` statement makes versioning a bit more confusing since you need to manage the version two places in the `go.mod` and I'm not sure it's necessary right now, although when versioning versions of software that do not use sem ver at all, it can definitely be useful.

**Comment by faddat** [comment]
You're right about the replace statements for sure.  I have to go and let others know ..... Thanks to you both @marbar3778 and @williambanfield for the lesson in go.sum ; I appreciate it.

**Pull request closed without merging**